shell/platform/darwin/ios/framework/Source/SemanticsObject.mm

```diff
@@ -223,17 +224,28 @@ - (id)accessibilityContainer {
 // private methods
 
 - (CGSize)contentSizeInternal {
-  CGRect result;
   const SkRect& rect = _semanticsObject.node.rect;
+  float x, y, width, height;
+  float scrollExtentMax = _semanticsObject.node.scrollExtentMax == INFINITY
+                              ? kScrollExtentMaxForInf + _semanticsObject.node.scrollPosition
+                              : _semanticsObject.node.scrollExtentMax;
   if (_semanticsObject.node.actions & flutter::kVerticalScrollSemanticsActions) {
-    result = CGRectMake(rect.x(), rect.y(), rect.width(),
-                        rect.height() + _semanticsObject.node.scrollExtentMax);
+    x = rect.x();
+    y = rect.y();
+    width = rect.width();
+    height = rect.height() + scrollExtentMax;
```

Comment:
nit: I liked the CGRectMake calls better fwiw.  It's a bit easier to maintain since it enforces all things are set before returning the result.